modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManagerTestCase.java

```diff
@@ -300,29 +284,51 @@ public void getOperationExecutorWithNotImplicitConfig()
     {
         when(muleContext.getRegistry().lookupObjects(ConfigurationProvider.class)).thenReturn(emptyList());
         makeExtension1ConfigurationNotImplicit();
-        discover();
 
         extensionsManager.getConfiguration(extensionModel1, event);
     }
 
     @Test
     public void registerTwoExtensionsWithTheSameNameButDifferentVendor()
     {
-        setDiscoverableExtensions(extensionModel2, extensionModel3WithRepeatedName);
-        List<RuntimeExtensionModel> extensionModels = extensionsManager.discoverExtensions(classLoader);
+        registerExtensions(extensionModel2, extensionModel3WithRepeatedName);
+        List<RuntimeExtensionModel> extensionModels = new ArrayList<>(extensionsManager.getExtensions());
         List<String> extensionNameList = extensionModels.stream().map(ExtensionModel::getName).distinct().collect(Collectors.toList());
         List<String> extensionVendorList = extensionModels.stream().map(ExtensionModel::getVendor).distinct().collect(Collectors.toList());
 
-        assertThat(extensionModels.size(), is(2));
-        assertThat(extensionNameList.size(), is(1));
+        assertThat(extensionModels.size(), is(3));
+        assertThat(extensionNameList.size(), is(2));
         assertThat(extensionVendorList.size(), is(2));
     }
 
     @Test
     public void registerSameExtensionMultipleTimesButOnlyKeepOne()
     {
-        setDiscoverableExtensions(extensionModel1, extensionModel1, extensionModel1);
-        assertThat(extensionsManager.discoverExtensions(classLoader).size(), is(1));
+        registerExtensions(extensionModel1, extensionModel1, extensionModel1);
+        assertThat(extensionsManager.getExtensions(), hasSize(3));
```

Comment:
I would change the test name to something like "ignoresRegisteringAlreadyRegisteredExtensions" or similar. I would also get the original number of extensions before calling registerExtensions and asserting the new value against it